### DISCUSSION TITLE
`Adjust pitch` setting is inverted

### DISCUSSION THREAD

**Discussion by exformation** [ppy/osu#27209] [Q&A]
https://github.com/ppy/osu/pull/24640#issuecomment-1950269050

"When turning on the Adjust pitch setting, the pitch of the track changes alongside the tempo"
This is just what audio tracks do by default when playback rate is changed.

In order to prevent it, you have to adjust the pitch down if the song is sped up, or adjust the pitch up if the song is slowed down. This is the default behavior of rate changing mods like DT/HT, because libbass does the fancy pitch correction with algorithms like phase vocoding or similar. 

Turning pitch correction off by turning `Adjust pitch` on is confusing, but maybe it wouldn't be for users that don't know any of this? I immediately thought it was a bug.

**Comment by OwenCMYK** [comment]
As a music producer, I understand what you mean, but for the average person, pitch and speed are thought of completely independently. Maybe "Change Pitch" or "Affect Pitch" might be better wording here, because "Adjust" could potentially imply that it was correcting/fixing it. But most people will likely think of this option as changing it from it's default, rather than turning off the correction.

**Comment by peppy** [comment]
the setting matches the name correctly. i'm not against changing the text if you come up with something better.

**Reply by OwenCMYK** [reply]
Personally I think "Affect Pitch" or maybe "Pitch Up/Down" for Double/Half Time respectively. But somebody else might think of a better idea.

**Reply by exformation** [reply]
"Disable Pitch Adjustment" would be my vote, because that's what the button is doing. Users could click it, hear the chipmunk voices on DT, and infer that pitch adjustment is the default behavior of DT/HT and they can disable that if they want.

**Reply by OwenCMYK** [reply]
I think it's better for DT to default to the song's original pitch (as it currently does). I also think "Disable Pitch Adjustment" is too complicated for the average person. If you're an audio engineer it makes sense, but for pretty much anyone else, that would be incredibly confusing. Because for the average person, the chipmunk effect *is* the "adjusted pitch". Also we don't want it to just be Nightcore without a beat, and since the beat doesn't play on song select, the difference would be hard to discern.

**Reply by frenzibyte** [reply]
> "Disable Pitch Adjustment" would be my vote, because that's what the button is doing. Users could click it, hear the chipmunk voices on DT, and infer that pitch adjustment is the default behavior of DT/HT and they can disable that if they want.

This name is directly the opposite of `Adjust Pitch`.

**Reply by OwenCMYK** [reply]
Yeah, opposite to how it is now. But the way it is now is intuitive to most people, so inverting it (while being technically correct) will be confusing to a lot of people.

**Reply by sz-da** [reply]
> "Disable Pitch Adjustment" would be my vote, because that's what the button is doing.

If you are considering inverting the functionality of the button, then using "Maintain Pitch" would be better than "Disable Pitch Adjustment" since "pitch adjustment" is sort of ambiguous

**Reply by exformation** [reply]
> > "Disable Pitch Adjustment" would be my vote, because that's what the button is doing. Users could click it, hear the chipmunk voices on DT, and infer that pitch adjustment is the default behavior of DT/HT and they can disable that if they want.
> 
> This name is directly the opposite of `Adjust Pitch`.

Yes, my posts are about how the setting is inverted and is called the opposite of what it's doing. The solutions to that are to keep the button the same and invert the wording or invert the button and improve the wording.

**Reply by exformation** [reply]
> > "Disable Pitch Adjustment" would be my vote, because that's what the button is doing.
> 
> If you are considering inverting the functionality of the button, then using "Maintain Pitch" would be better than "Disable Pitch Adjustment" since "pitch adjustment" is sort of ambiguous

This sounds good.

**Reply by frenzibyte** [reply]
Are you sure this makes more sense to you? The term "maintain" means to keep pitch the same, i.e. not adjusting it. If we were to flip the setting, then the default value would be "on", which is to not make the track sound like "chipumunk voices" as you say it.

**Reply by exformation** [reply]
If the text was "Maintain Pitch" then the button would have to be on by default, because that's the default behavior of DT/HT, so this would be the option of inverting the button and improving wording.
